src/index.js

```diff
@@ -46,47 +45,33 @@ module.exports = {
     'id-length': ['error', {
       exceptions: ['_', 'e', 'i']
     }],
+    'id-match': ['error', '^_$|^[a-zA-Z][a-zA-Z0-9]*$|^[A-Z][A-Z0-9]*_[A-Z0-9]+$'],
```

Comment:
This still doesn't seem to handle uppercase names with more than one underscore.